jf_agent/jira_download.py

```diff
@@ -273,30 +273,29 @@ def download_worklogs(jira_connection, issue_ids):
     return {'existing': updated, 'deleted': []}
 
 
-def _jira_user_key(user_dict, gdpr_active):
-    if gdpr_active:
-        return user_dict['accountId']
-    else:
-        return user_dict['key']
+def _jira_user_key(user_dict):
+    return user_dict.get('key', user_dict['accountId'])
 
 
 def _search_all_users(jira_connection, gdpr_active):
     jira_users = {}
     start_at = 0
+
+    # iterate through pages of results
     while True:
-        # Search for '%' or '@' so we get results from cloud or
-        # on-prem, the latter being what on-prem appears to use as
-        # a wildcard
-        users = _search_users(
-            jira_connection, gdpr_active, query='%', start_at=start_at, include_inactive=True
-        ) or _search_users(
-            jira_connection, gdpr_active, query='@', start_at=start_at, include_inactive=True
-        )
+        users = None
+
+        # different jira versions / different times, the way to list all users has changed. Try a few.
+        for q in [None, '', '%', '@']:
```

Comment:
should this `for` loop be outside the `while True`?  so it paginates through `q=None`, then through `q=''`, then `q='%'` etc?

Reply:
I thought the code was simpler this way, but you're right it's a little wasteful to be trying all the query strings on each page iteration.